nncf/pruning/filter_pruning/algo.py

```diff
@@ -83,19 +98,178 @@ def statistics(self, quickly_collected_only=False):
     def freeze(self):
         self.frozen = True
 
+    def _init_pruned_modules_params(self):
+        def get_in_out_channels(module):
+            in_channels, out_channels = None, None
+            if isinstance(module, (nn.Conv1d, nn.Conv2d, nn.Conv3d, nn.ConvTranspose1d, nn.ConvTranspose2d,
```

Comment:
It seems you already have this code in utils.py. I propose to make it more granular and reuse here.

Reply:
These functions have clear objectives and aren't similar enough to be united. I propose to leave them separately to keep them simple.